src/_pytest/nodes.py

```diff
@@ -553,11 +553,9 @@ def _gethookproxy(self, fspath: py.path.local):
         remove_mods = pm._conftest_plugins.difference(my_conftestmodules)
         if remove_mods:
             # one or more conftests are not in use at this fspath
-            proxy = FSHookProxy(pm, remove_mods)
-        else:
-            # all plugins are active for this fspath
-            proxy = self.config.hook
-        return proxy
+            return FSHookProxy(pm, remove_mods)
+        # all plugins are active for this fspath
+        return self.config.hook
```

Comment:
I would keep the `if: else:` here for human readability (it's easier for a human to see at a glance that these two paths are disparate, and more resistant to errors during refactoring which would cause bugs)